nexus/tests/integration_tests/instances.rs

```diff
@@ -3149,6 +3153,83 @@ async fn test_instance_create_in_silo(cptestctx: &ControlPlaneTestContext) {
         .expect("Failed to parse instance");
 }
 
+/// Test that appropriate OPTE V2P mappings are created and deleted.
+#[nexus_test]
+async fn test_instance_v2p_mappings(cptestctx: &ControlPlaneTestContext) {
+    let client = &cptestctx.external_client;
+
+    populate_ip_pool(&client, "default", None).await;
+    create_organization(&client, ORGANIZATION_NAME).await;
+    create_project(client, ORGANIZATION_NAME, PROJECT_NAME).await;
+
+    // Add a few more sleds
+    let nsleds = 3;
+    let mut additional_sleds = Vec::with_capacity(nsleds);
+    for _ in 0..nsleds {
+        let sa_id = Uuid::new_v4();
+        let log =
+            cptestctx.logctx.log.new(o!( "sled_id" => sa_id.to_string() ));
+        let addr = cptestctx.server.get_http_server_internal_address().await;
+        let update_directory = std::path::Path::new("/should/not/be/used");
+        additional_sleds.push(
+            start_sled_agent(log, addr, sa_id, &update_directory)
+                .await
+                .unwrap(),
+        );
+    }
+
+    // Create an instance.
+    let instance_name = "test-instance";
+    let instance =
+        create_instance(client, ORGANIZATION_NAME, PROJECT_NAME, instance_name)
+            .await;
+
+    // Validate that every sled (except the instance's sled) now has a V2P
+    // mapping for this instance
+    let apictx = &cptestctx.server.apictx();
+    let nexus = &apictx.nexus;
+    let datastore = nexus.datastore();
+    let opctx =
+        OpContext::for_tests(cptestctx.logctx.log.new(o!()), datastore.clone());
+
+    let (.., db_instance) = LookupPath::new(&opctx, &datastore)
+        .instance_id(instance.identity.id)
+        .fetch()
+        .await
+        .unwrap();
+
+    let mut sled_agents: Vec<&Arc<SledAgent>> =
+        additional_sleds.iter().map(|x| &x.sled_agent).collect();
+    sled_agents.push(&cptestctx.sled_agent.sled_agent);
+
+    for sled_agent in &sled_agents {
+        let v2p_mappings = sled_agent.v2p_mappings.lock().await;
+        if sled_agent.id != db_instance.runtime().sled_id {
+            assert!(!v2p_mappings.is_empty());
```

Comment:
Could we test the contents of the mappings? I don't see any other location that we ensure we've really provided the right data to OPTE.

Reply:
Good call, done in 210adc9a